### PR TITLE
Tesla gun thing

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

It now applies a debilitating plasma draining affect and can be ordered in req
https://streamable.com/0bdxjk

## Why It's Good For The Game

@Tennessee116 you can make doom BFG 9000 using this
## Changelog
:cl:
add: TG beams
add: So far admin only tesla gun that shoots and damages nearby enemies with its projectiles as it gets close
/:cl:

### CODE REVIEW

File: code/datums/beam.dm [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	var/datum/beam/newbeam = new(src,BeamTarget,icon,icon_state,time,maxdistance,beam_type,beam_sleep_time)
+	INVOKE_ASYNC(newbeam, /datum/beam/.proc/Start)
+	return newbeam
+
+/proc/zap_beam(atom/source, zap_range, damage, list/blacklistmobs)
+	var/list/mobs = list()
+	for(var/mob/living/carbon/xenomorph/beno in oview(zap_range, source))
+		mobs += beno
+	for(var/xeno in mobs)
+		var/mob/living/carbon/xenomorph/living = xeno
```
Comment (TiviPlus) [inline code/datums/beam.dm:171]: need help with making this less shit

**Comment by Tennessee116** [comment]
> ## Why It's Good For The Game
> @Tennessee116 you can make doom BFG 9000 using this

People making things because it's good for me?

Now I feel lubbed.

image

File: code/datums/beam.dm (hunk 1 above)
Comment (TiviPlus) [reply code/datums/beam.dm]: made it less shit but theres still something i need to look into

Comment (TiviPlus) [reply code/datums/beam.dm]: Okay thanks byond I guess, this is good now

**Pull request merged**